### ISSUE TITLE
Insufficient help for declare

### ISSUE THREAD

**Issue by milancermak** [xJonathanLEI/starkli#12]
Running `starkli declare` with no args, just to see the help message yields

```
starkli declare
error: the following required arguments were not provided:
  <FILE>

Usage: starkli declare --account <ACCOUNT> --rpc <RPC> --network <NETWORK> <FILE>

For more information, try '--help'.
```

But for `declare` to work, I must pass in the `--keystore` as well. This is not obvious just from the help message. When I ran it without `--keystore` the output was

```
Error: no valid signer option provided
```

which did lead me to the right solution, though I initially used `--signer` instead of `--keystore` because I'm new to the tool and the error message mentions "signer" 🙈

Can the initial help message of `declare` be improved to mention the command needs a signer as well?

**Comment by xJonathanLEI** [comment]
Thanks for the feedback. The current initial error message there is automatically generated by `clap`. Not sure if we can hack into that one. Can check on that though.

The 2nd one can certainly be improved. Currently it doesn't mention `--keystore` at all because it's not the only accepted signer option. For example you can pass `--private-key` and it will happily use that to sign your transactions. That said, it's certainly true that we can explain better what's wrong and even link to the docs (non-existent for now).

**Comment by barretodavid** [comment]
btw, I think `--signer` is a better name than `--keystore` for the flag. I had a similar confusion first time I used it

**Comment by xJonathanLEI** [comment]
Using `--signer` for `--keystore` is appropriate only if keystore is the only signer implementation. It's not the case. We already also have plain text private key support, and this will be more apparent once hardware wallet support lands. Using `--signer` to mean keystore would be very confusing.

**Comment by barretodavid** [comment]
Ok makes sense.

I noticed something else in this command. When you display the help, the `--account` flag is marked both as required and also under `[OPTIONS]`. 

```
$ starkli declare --help 
>>>
                                                                     
Declare a contract class

Usage: starkli declare [OPTIONS] --account <ACCOUNT> <FILE>

Arguments:
  <FILE>  Path to contract artifact file

Options:
      --rpc <RPC>
          Starknet JSON-RPC endpoint [env: STARKNET_RPC=https://starknet-goerli.g.alchemy.com/v2/V0WIoJ3TeQ995GdeHM5kZcilzzcIg_0U]
      --network <NETWORK>
          Starknet network [env: STARKNET_NETWORK=] [possible values: mainnet, goerli-1, goerli-2, integration]
      --keystore <KEYSTORE>
          Path to keystore JSON file
      --keystore-password <KEYSTORE_PASSWORD>
          Supply keystore password from command line option instead of prompt
      --private-key <PRIVATE_KEY>
          Private key in hex in plain text
      --compiler-version <COMPILER_VERSION>
          Statically-linked Sierra compiler version [possible values: 1.1.0, 2.0.1]
      --casm-hash <CASM_HASH>
          Override Sierra compilation and use CASM hash directly
      --account <ACCOUNT>
          Path to account config JSON file [env: STARKNET_ACCOUNT=]
      --estimate-only
          Only estimate transaction fee without sending transaction
      --watch
          Wait for the transaction to confirm
  -h, --help
          Print help
```

The description under options says that it can be set with the environment variable `STARKNET_ACCOUNT` which I did but if I try to run the command I get an error

```
$ starkli declare target/dev/cairo1_v2_Ownable.sierra.json
>>>

error: the following required arguments were not provided:
  --account <ACCOUNT>

Usage: starkli declare --account <ACCOUNT> --rpc <RPC> <FILE>

For more information, try '--help'.
```

Let me know if you want me to open a new issue to track it.

**Comment by barretodavid** [comment]
Also, is it possible to declare the location of the keystore file as an env variable? The help seems to imply that it has to be passed as a flag but an env var would be more convenient I think

**Comment by milancermak** [comment]
@xJonathanLEI couldn't `--signer` be a catch-all arg tho? The code could do a check if the value is a keystore file or a plaintext file or a private key and act appropriately no?

**Comment by milancermak** [comment]
> Also, is it possible to declare the location of the keystore file as an env variable?

Yes, via `STARKNET_KEYSTORE`.

**Comment by barretodavid** [comment]
@milancermak I was able to find out the right env variable name but notice that the help doesn't say it. Same for `--account`

**Comment by xJonathanLEI** [comment]
> @milancermak I was able to find out the right env variable name but notice that the help doesn't say it

The help text mentions `STARKNET_KEYSTORE` now: https://github.com/xJonathanLEI/starkli/commit/ddfb5aaa3a0642cf7e2df783104fcf7e9cd2d9a2.

**Comment by xJonathanLEI** [comment]
Just pushed https://github.com/xJonathanLEI/starkli/commit/2d3cebd0dc6a66ada9c248390431b6dae470eebc. Running any command that requires a signer without supplying a signer now prints this instead:

```console
Error: no valid signer option provided. Consider using a keystore by providing a --keystore option.

For more information, see: https://book.starkli.rs/signers
```

Now it gives more helpful instructions, and also points to the relevant page in the book.

Closing this issue optimistically now as I consider it resolved. Feel free to re-open if you disagree.
